dashboard/src/pages/BuildDetails/BuildDetails.tsx

```diff
@@ -40,8 +43,12 @@ const BuildDetailsPage = (): JSX.Element => {
           return {
             ...previousParams,
             tableFilter: {
-              bootsTable: previousParams.tableFilter.bootsTable,
-              buildsTable: previousParams.tableFilter.buildsTable,
+              bootsTable:
+                previousParams.tableFilter?.bootsTable ??
+                zTableFilterInfoDefault.bootsTable,
+              buildsTable:
+                previousParams.tableFilter?.buildsTable ??
+                zTableFilterInfoDefault.buildsTable,
```

Comment:
On some other files (hardwareDetails boots and tests tabs and treeDetails boots tab), the change in the table filter is passed such as 
```
tableFilter: {
    ...previousParams.tableFilter,
    bootsTable: newFilter,
}
```
that way is a bit cleaner, you could change it to that format if they show the same behavior (which in my tests they do)

Reply:
Done!